server/infinispan/src/test/java/org/hawkular/apm/server/infinispan/InfinispanPropertyCacheTest.java

```diff
@@ -0,0 +1,204 @@
+
+        cache.store(null, new Property<>("id", "foo"));
+        CountingObserver observer = new CountingObserver();
+        cache.addObserver(null, "id", observer);
+        cache.store(null, new Property<>("id", "bar"));
+        Assert.assertEquals(1, observer.updatedCounter.intValue());
+
+        cache.remove(null, "id");
+        Assert.assertNull(cache.get(null, "id"));
+    }
+
+    @Test
+    public void testSystemProp() {
+        InfinispanPropertyCache cache = createCache();
+
+        System.setProperty("strProp", "foo");
+        Assert.assertEquals("foo", cache.get(null, "strProp").getValue());
+        System.clearProperty("strProp");
+
+        System.setProperty("intProp", "2");
+        Assert.assertEquals("2", cache.get(null, "intProp").getValue());
+        System.clearProperty("intProp");
+    }
+
+    @Test
+    public void testObservableCreated() throws CacheException {
+        InfinispanPropertyCache cache = createCache();
+
+        CountingObserver observer = new CountingObserver();
+        cache.addObserver(null, "id1", observer);
+
+        cache.store(null, new Property<>("id1", 2));
+        Assert.assertEquals(1, observer.createdCounter.intValue());
+
+        cache.store(null, new Property<>("id1", 22));
+        Assert.assertEquals(1, observer.createdCounter.intValue());
+    }
+
+    @Test
+    public void testObservableUpdated() throws CacheException {
+        InfinispanPropertyCache cache = createCache();
+
+        CountingObserver observer = new CountingObserver();
+        cache.addObserver(null, "id1", observer);
+
+        cache.store(null, new Property<>("id1", 2));
+        Assert.assertEquals(0, observer.updatedCounter.intValue());
+
+        cache.store(null, new Property<>("id1", 2));
+        Assert.assertEquals(0, observer.updatedCounter.intValue());
+
+        cache.store(null, new Property<>("id1", 111));
+        Assert.assertEquals(1, observer.updatedCounter.intValue());
+    }
+
+    @Test
+    public void testObservableRemoved() throws CacheException {
+        InfinispanPropertyCache cache = createCache();
+
+        CountingObserver observer = new CountingObserver();
+        cache.addObserver(null, "id1", observer);
+
+        cache.store(null, new Property<>("id1", 2));
+        Assert.assertEquals(0, observer.removedCounter.intValue());
+
+        cache.store(null, new Property<>("id1", 2));
+        Assert.assertEquals(0, observer.removedCounter.intValue());
+
+        cache.store(null, new Property<>("id1", 111));
+        Assert.assertEquals(0, observer.removedCounter.intValue());
+
+        cache.remove(null, "id1");
+        Assert.assertEquals(1, observer.removedCounter.intValue());
+        cache.remove(null, "id1");
+        Assert.assertEquals(1, observer.removedCounter.intValue());
+
+        cache.store(null, new Property<>("id1", 222));
+        cache.remove(null, "id1");
+        Assert.assertEquals(1, observer.removedCounter.intValue());
```

Comment:
I think this should be 2, but possibly isn't at the moment due to the removal of the property also removing the observer. Once that is sorted, I think this will need to be 2.